codesign/lib/src/file_codesign_visitor.dart

```diff
@@ -341,6 +341,12 @@ update these file paths accordingly.
       match = _notarytoolRequestPattern.firstMatch(combinedOutput);
 
       if (match == null) {
+        if (result.exitCode >= 0 && result.exitCode <= 255) {
```

Comment:
I think actually before you even use the regex, after line 337, you could do:

```dart
if (result.exitCode != 0) {
  throw CodesignException('Command "${args.join(' ')}" failed with exit code ${result.exitCode}\nStdout: ${result.stdout}\nStderr: ${result.stderr}');
}
```

There's no point in continuing to run if you have a non-zero exit code, and this requires manual intervention.

Reply:
Sure, added a check for bad exit code <0 or exit code >255 on Mac.